cmd/kwil-admin/cmds/snapshot/create.go

```diff
@@ -0,0 +1,145 @@
+package snapshot
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	createLongExplain = `Creates a snapshot of the database. The command is used during network migration to get the state of the KwilDB.
+		It interacts directly with the Postgres server without intervening the kwild node.`
+)
+
+func createCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create <db-name> <db-user> <db-host> <db-port> <snapshot-dir>",
```

Comment:
Could the db config be optional flags instead?

Defaults would likely be:
name: postgres
user: postgres
host: localhost
port: 5432

Also, you forgot password (which the default would be "password")

Reply:
I think password and port need to be optional. If we're connecting with unix sockets, port has to be empty, and if and "trust" auth in pg_hba.conf, password also should be empty.

Needing to have empty strings for some if these, together with sensible defaults that will work with many installs including our docker configs, it suggests flags would be preferable to positionals.

Is this why you were asking about URI instead? Yeah, maybe. I like to be able to just specify e.g. host, and go with the defaults for the other pieces though, and I can't do that with a URI.

Reply:
Yeah, this is sort've why I was asking about URI. IMO it feels more flexible as a user, but that's just because it is what I have used before.

Reply:
Fair enough.   I find URI overly restrictive and requires you to assemble it (knowing all the pieces) if you just want to set user for example.  I see where you're coming from though.  I personally think we should not wire up URI.  Also just about all the `pg_` commands take the pieces with flags, as do most examples for the commands.